### ISSUE TITLE
[Bug]: segmentation fault

### ISSUE THREAD

**Issue by StarrySeas1** [run-llama/llama_index#11598]
### Bug Description

ERROR: [1]    33056 segmentation fault
Execute test cases
`
from llama_index.embeddings.huggingface import HuggingFaceEmbedding
embed_model = HuggingFaceEmbedding(model_name="/Users/mac/Documents/code/PaperCraftAI/ckpt/bge-base-zh-v1.5")
`

OS: macos 14.3.1 (23D60)

Running environment configuration:
|Name           |         Version          |         Build  | Channel
| :----| ----: | :----: |:----: |
accelerate|0.27.2|pypi_0|pypi
aiohttp|3.9.3|pypi_0|pypi
aiosignal|1.3.1|pypi_0|pypi
annotated-types|0.6.0|pypi_0|pypi
anyio|4.3.0|pypi_0|pypi
appnope|0.1.4|pyhd8ed1ab_0|conda-forge
asgiref|3.7.2|pypi_0|pypi
asttokens|2.4.1|pyhd8ed1ab_0|conda-forge
async-timeout|4.0.3|pypi_0|pypi
attrs|23.2.0|pypi_0|pypi
backcall|0.2.0|pyh9f0ad1d_0|conda-forge
backoff|2.2.1|pypi_0|pypi
bcrypt|4.1.2|pypi_0|pypi
beautifulsoup4|4.12.3|pypi_0|pypi
blas|1.0|mkl|defaults
bs4|0.0.2|pypi_0|pypi
build|1.1.1|pypi_0|pypi
bzip2|1.0.8|h6c40b1e_5|defaults
c-ares|1.19.1|h6c40b1e_0|defaults
ca-certificates|2023.12.12|hecd8cb5_0|defaults
cachetools|5.3.3|pypi_0|pypi
certifi|2024.2.2|py39hecd8cb5_0|defaults
chardet|5.2.0|pypi_0|pypi
charset-normalizer|3.3.2|pypi_0|pypi
chroma-hnswlib|0.7.3|pypi_0|pypi
chromadb|0.4.24|pypi_0|pypi
click|8.1.7|pypi_0|pypi
coloredlogs|15.0.1|py39hecd8cb5_2|defaults
comm|0.2.1|pyhd8ed1ab_0|conda-forge
cpuonly|2.0|0|pytorch
dataclasses-json|0.6.4|pypi_0|pypi
debugpy|1.6.7|py39hcec6c5f_0|defaults
decorator|5.1.1|pyhd8ed1ab_0|conda-forge
deprecated|1.2.14|pypi_0|pypi
dirtyjson|1.0.8|pypi_0|pypi
distro|1.9.0|pypi_0|pypi
exceptiongroup|1.2.0|pypi_0|pypi
executing|2.0.1|pyhd8ed1ab_0|conda-forge
fastapi|0.110.0|pypi_0|pypi
ffmpeg|4.3|h0a44026_0|pytorch
filelock|3.13.1|pypi_0|pypi
flatbuffers|23.5.26|pypi_0|pypi
freetype|2.12.1|hd8bbffd_0|defaults
frozenlist|1.4.1|pypi_0|pypi
fsspec|2024.2.0|pypi_0|pypi
gettext|0.21.0|he85b6c0_1|defaults
gmp|6.2.1|he9d5cce_3|defaults
gmpy2|2.1.2|py39hd5de756_0|defaults
gnutls|3.6.15|hed9c0bf_0|defaults
google-auth|2.28.1|pypi_0|pypi
googleapis-common-protos|1.62.0|pypi_0|pypi
greenlet|3.0.3|pypi_0|pypi
grpcio|1.62.0|pypi_0|pypi
h11|0.14.0|pypi_0|pypi
httpcore|1.0.4|pypi_0|pypi
httptools|0.6.1|pypi_0|pypi
httpx|0.27.0|pypi_0|pypi
huggingface-hub|0.21.3|pypi_0|pypi
humanfriendly|10.0|py39hecd8cb5_1|defaults
icu|73.1|hcec6c5f_0|defaults
idna|3.6|pypi_0|pypi
importlib-metadata|6.11.0|pypi_0|pypi
importlib-resources|6.1.2|pypi_0|pypi
importlib_metadata|7.0.1|hd8ed1ab_0|conda-forge
instructorembedding|1.0.1|pypi_0|pypi
intel-openmp|2023.1.0|ha357a0b_43548|defaults
ipykernel|6.29.3|pyh3cd1d5f_0|conda-forge
ipython|8.12.0|pyhd1c38e8_0|conda-forge
jedi|0.19.1|pyhd8ed1ab_0|conda-forge
jinja2|3.1.3|pypi_0|pypi
joblib|1.3.2|pypi_0|pypi
jpeg|9e|h6c40b1e_1|defaults
jupyter_client|8.6.0|pyhd8ed1ab_0|conda-forge
jupyter_core|5.7.1|py39h6e9494a_0|conda-forge
krb5|1.20.1|h428f121_1|defaults
kubernetes|29.0.0|pypi_0|pypi
lame|3.100|h1de35cc_0|defaults
lcms2|2.12|hf1fd2bf_0|defaults
lerc|3.0|he9d5cce_0|defaults
libcurl|8.5.0|hf20ceda_0|defaults
libcxx|14.0.6|h9765a3e_0|defaults
libdeflate|1.17|hb664fd8_1|defaults
libedit|3.1.20230828|h6c40b1e_0|defaults
libev|4.33|h9ed2024_1|defaults
libffi|3.4.4|hecd8cb5_0|defaults
libiconv|1.16|hca72f7f_2|defaults
libidn2|2.3.4|h6c40b1e_0|defaults
libnghttp2|1.57.0|h9beae6a_0|defaults
libpng|1.6.39|h6c40b1e_0|defaults
libprotobuf|3.20.3|hfff2838_0|defaults
libsodium|1.0.18|hbcb3906_1|conda-forge
libssh2|1.10.0|h04015c4_2|defaults
libtasn1|4.19.0|h6c40b1e_0|defaults
libtiff|4.5.1|hcec6c5f_0|defaults
libunistring|0.9.10|h9ed2024_0|defaults
libwebp-base|1.3.2|h6c40b1e_0|defaults
libxml2|2.10.4|h1bd7e62_1|defaults
llama-index|0.10.15|pypi_0|pypi
llama-index-agent-openai|0.1.5|pypi_0|pypi
llama-index-cli|0.1.7|pypi_0|pypi
llama-index-core|0.10.15|pypi_0|pypi
llama-index-embeddings-huggingface|0.1.4|pypi_0|pypi
llama-index-embeddings-instructor|0.1.3|pypi_0|pypi
llama-index-embeddings-openai|0.1.6|pypi_0|pypi
llama-index-indices-managed-llama-cloud|0.1.3|pypi_0|pypi
llama-index-legacy|0.9.48|pypi_0|pypi
llama-index-llms-openai|0.1.7|pypi_0|pypi
llama-index-multi-modal-llms-openai|0.1.4|pypi_0|pypi
llama-index-program-openai|0.1.4|pypi_0|pypi
llama-index-question-gen-openai|0.1.3|pypi_0|pypi
llama-index-readers-file|0.1.6|pypi_0|pypi
llama-index-readers-llama-parse|0.1.3|pypi_0|pypi
llama-index-vector-stores-chroma|0.1.5|pypi_0|pypi
llama-parse|0.3.5|pypi_0|pypi
llamaindex-py-client|0.1.13|pypi_0|pypi
llvm-openmp|14.0.6|h0dcd299_0|defaults
lz4-c|1.9.4|hcec6c5f_0|defaults
markupsafe|2.1.5|pypi_0|pypi
marshmallow|3.21.0|pypi_0|pypi
matplotlib-inline|0.1.6|pyhd8ed1ab_0|conda-forge
mkl|2023.1.0|h8e150cf_43560|defaults
mkl-service|2.4.0|py39h6c40b1e_1|defaults
mkl_fft|1.3.8|py39h6c40b1e_0|defaults
mkl_random|1.2.4|py39ha357a0b_0|defaults
mmh3|4.1.0|pypi_0|pypi
monotonic|1.6|pypi_0|pypi
mpc|1.1.0|h6ef4df4_1|defaults
mpfr|4.0.2|h9066e36_1|defaults
mpmath|1.3.0|py39hecd8cb5_0|defaults
multidict|6.0.5|pypi_0|pypi
mypy-extensions|1.0.0|pypi_0|pypi
ncurses|6.4|hcec6c5f_0|defaults
nest-asyncio|1.6.0|pyhd8ed1ab_0|conda-forge
nettle|3.7.3|h230ac6f_1|defaults
networkx|3.2.1|pypi_0|pypi
nltk|3.8.1|pypi_0|pypi
numpy|1.26.4|py39h827a554_0|defaults
numpy-base|1.26.4|py39ha186be2_0|defaults
oauthlib|3.2.2|pypi_0|pypi
onnxruntime|1.17.1|pypi_0|pypi
openai|1.13.3|pypi_0|pypi
openh264|2.1.1|h8346a28_0|defaults
openjpeg|2.4.0|h66ea3da_0|defaults
openssl|3.0.13|hca72f7f_0|defaults
opentelemetry-api|1.23.0|pypi_0|pypi
opentelemetry-exporter-otlp-proto-common|1.23.0|pypi_0|pypi
opentelemetry-exporter-otlp-proto-grpc|1.23.0|pypi_0|pypi
opentelemetry-instrumentation|0.44b0|pypi_0|pypi
opentelemetry-instrumentation-asgi|0.44b0|pypi_0|pypi
opentelemetry-instrumentation-fastapi|0.44b0|pypi_0|pypi
opentelemetry-proto|1.23.0|pypi_0|pypi
opentelemetry-sdk|1.23.0|pypi_0|pypi
opentelemetry-semantic-conventions|0.44b0|pypi_0|pypi
opentelemetry-util-http|0.44b0|pypi_0|pypi
orjson|3.9.15|pypi_0|pypi
overrides|7.7.0|pypi_0|pypi
packaging|23.2|pyhd8ed1ab_0|conda-forge
pandas|2.2.1|pypi_0|pypi
parso|0.8.3|pyhd8ed1ab_0|conda-forge
pexpect|4.9.0|pyhd8ed1ab_0|conda-forge
pickleshare|0.7.5|py_1003|conda-forge
pillow|10.2.0|py39h6c40b1e_0|defaults
pip|24.0|pypi_0|pypi
platformdirs|4.2.0|pyhd8ed1ab_0|conda-forge
posthog|3.4.2|pypi_0|pypi
prompt-toolkit|3.0.42|pyha770c72_0|conda-forge
prompt_toolkit|3.0.42|hd8ed1ab_0|conda-forge
protobuf|4.25.3|pypi_0|pypi
psutil|5.9.8|py39ha09f3b3_0|conda-forge
ptyprocess|0.7.0|pyhd3deb0d_0|conda-forge
pulsar-client|3.4.0|pypi_0|pypi
pure_eval|0.2.2|pyhd8ed1ab_0|conda-forge
pyasn1|0.5.1|pypi_0|pypi
pyasn1-modules|0.3.0|pypi_0|pypi
pydantic|2.6.3|pypi_0|pypi
pydantic-core|2.16.3|pypi_0|pypi
pygments|2.17.2|pyhd8ed1ab_0|conda-forge
pymupdf|1.23.26|pypi_0|pypi
pymupdfb|1.23.22|pypi_0|pypi
pypdf|4.1.0|pypi_0|pypi
pypika|0.48.9|pypi_0|pypi
pyproject-hooks|1.0.0|pypi_0|pypi
python|3.9.18|h5ee71fb_0|defaults
python-dateutil|2.9.0|pyhd8ed1ab_0|conda-forge
python-dotenv|1.0.1|pypi_0|pypi
python-flatbuffers|2.0|pyhd3eb1b0_0|defaults
python_abi|3.9|2_cp39|conda-forge
pytorch-mutex|1.0|cpu|pytorch
pytz|2024.1|pypi_0|pypi
pyyaml|6.0.1|pypi_0|pypi
pyzmq|24.0.1|py39hed8f129_1|conda-forge
readline|8.2|hca72f7f_0|defaults
regex|2023.12.25|pypi_0|pypi
requests|2.31.0|py39hecd8cb5_1|defaults
requests-oauthlib|1.3.1|pypi_0|pypi
rsa|4.9|pypi_0|pypi
safetensors|0.4.2|pypi_0|pypi
scikit-learn|1.4.1.post1|pypi_0|pypi
scipy|1.12.0|pypi_0|pypi
sentence-transformers|2.5.1|pypi_0|pypi
setuptools|68.2.2|py39hecd8cb5_0|defaults
six|1.16.0|pyh6c4a22f_0|conda-forge
sniffio|1.3.1|pypi_0|pypi
soupsieve|2.5|pypi_0|pypi
sqlalchemy|2.0.27|pypi_0|pypi
sqlite|3.41.2|h6c40b1e_0|defaults
stack_data|0.6.2|pyhd8ed1ab_0|conda-forge
starlette|0.36.3|pypi_0|pypi
sympy|1.12|py39hecd8cb5_0|defaults
tbb|2021.8.0|ha357a0b_0|defaults
tenacity|8.2.3|pypi_0|pypi
threadpoolctl|3.3.0|pypi_0|pypi
tiktoken|0.6.0|pypi_0|pypi
tk|8.6.12|h5d9f67b_0|defaults
tokenizers|0.15.2|pypi_0|pypi
tomli|2.0.1|pypi_0|pypi
torch|2.1.2|pypi_0|pypi
torchaudio|2.1.2|pypi_0|pypi
torchvision|0.16.2|pypi_0|pypi
tornado|6.4|py39ha09f3b3_0|conda-forge
tqdm|4.66.2|pypi_0|pypi
traitlets|5.14.1|pyhd8ed1ab_0|conda-forge
transformers|4.38.2|pypi_0|pypi
typer|0.9.0|pypi_0|pypi
typing-extensions|4.10.0|pypi_0|pypi
typing-inspect|0.9.0|pypi_0|pypi
typing_extensions|4.9.0|py39hecd8cb5_1|defaults
tzdata|2024.1|pypi_0|pypi
urllib3|2.2.1|pypi_0|pypi
uvicorn|0.27.1|pypi_0|pypi
uvloop|0.19.0|pypi_0|pypi
watchfiles|0.21.0|pypi_0|pypi
wcwidth|0.2.13|pyhd8ed1ab_0|conda-forge
websocket-client|1.7.0|pypi_0|pypi
websockets|12.0|pypi_0|pypi
wheel|0.41.2|py39hecd8cb5_0|defaults
wrapt|1.16.0|pypi_0|pypi
xz|5.4.6|h6c40b1e_0|defaults
yarl|1.9.4|pypi_0|pypi
zeromq|4.3.5|hcec6c5f_0|defaults
zipp|3.17.0|pyhd8ed1ab_0|conda-forge
zlib|1.2.13|h4dc903c_0|defaults
zstd|1.5.5|hc035e20_0|defaults

### Version

llama-index               0.10.15

### Steps to Reproduce

To keep your environment consistent, execute the following code.
`
from llama_index.embeddings.huggingface import HuggingFaceEmbedding
embed_model = HuggingFaceEmbedding(model_name="/Users/mac/Documents/code/PaperCraftAI/ckpt/bge-base-zh-v1.5")
`

**Comment by logan-markewich** [comment]
I don't think you are going to get much help here with this.

Its just calling AutModel.from_pretrained() and AutokTokenizer.from_pretrained() with the path you gave.
